src/main/java/com/google/jenkins/plugins/storage/ExpiringBucketLifecycleManager.java

```diff
@@ -25,154 +25,165 @@
 import org.kohsuke.stapler.DataBoundConstructor;
 
 import com.google.api.services.storage.model.Bucket;
-import com.google.common.base.Objects;
 import com.google.common.collect.Lists;
 
 import hudson.Extension;
 
 /**
- * A simple implementation of the bucket lifecycle extension point
- * that surfaces object expiration (aka TTL).
+ * A simple implementation of the bucket lifecycle extension point that surfaces
+ * object expiration (aka TTL).
  */
 public class ExpiringBucketLifecycleManager
-    extends AbstractBucketLifecycleManager {
-  private static final Logger logger =
-      Logger.getLogger(ExpiringBucketLifecycleManager.class.getName());
-
-  /**
-   * Construct the simple lifecycle manager from a TLL and the
-   * common properties.
-   */
-  @DataBoundConstructor
-  public ExpiringBucketLifecycleManager(String bucket,
-      @Nullable UploadModule module, Integer ttl,
-      // Legacy arguments for backwards compatibility
-      @Deprecated @Nullable String bucketNameWithVars,
-      @Deprecated @Nullable Integer bucketObjectTTL) {
-    super(Objects.firstNonNull(bucket, bucketNameWithVars), module);
-
-    this.bucketObjectTTL = Objects.firstNonNull(ttl, bucketObjectTTL);
-  }
-
-  /**
-   * {@inheritDoc}
-   */
-  @Override
-  public String getDetails() {
-    return Messages.ExpiringBucketLifecycleManager_DetailsMessage(
-        getTtl());
-  }
-
-  /**
-   * {@inheritDoc}
-   */
-  @Override
-  protected Bucket checkBucket(Bucket bucket)
-      throws InvalidAnnotationException {
-    Bucket.Lifecycle lifecycle = bucket.getLifecycle();
-    if (lifecycle == null) {
-      throw new InvalidAnnotationException(bucket);
-    }
+        extends AbstractBucketLifecycleManager {
+
+    private static final Logger logger
+            = Logger.getLogger(ExpiringBucketLifecycleManager.class.getName());
 
-    if (lifecycle.getRule().size() != 1) {
-      // TODO(mattmoor): Consider allowing this plugin to augment, rather
-      // than replace existing lifecycle rules.
-      if (lifecycle.getRule().size() > 1) {
-        logger.log(WARNING, "Found complex lifecycle rule on: " +
-            bucket.getName());
-      }
-      throw new InvalidAnnotationException(bucket);
+    /**
+     * Construct the simple lifecycle manager from a TLL and the common
+     * properties.
+     */
+    public static <T> T firstNonNull(T... objs) {
```

Comment:
Could we avoid this code repetition?

Also, the comment is in the wrong place.

Reply:
Comment in same place as it's was before.

I can't avoid code repetition without rewriting plugin totally

Reply:
A function that is repeated like that should be extracted as a library function. In this case, however, I believe the library function already exists.